### ISSUE TITLE
Fix ODIS input validation checks in common package

### ISSUE THREAD

**Issue by alecps** [celo-org/celo-monorepo#6824]
Some of the input validation checks simply check for the existence of a field but do not validate that field in any way. This can cause unexpected errors to be thrown if invalid input is sent in a request. We should improve these checks so that we always return 400 when we get bad input

**Comment by aslawson** [comment]
@alecps @codyborn  this says it's blocked by a p2 unit test. is this super high priority?

**Comment by codyborn** [comment]
I don't think it's a hard blocker. @alecps WDYT?

**Comment by alecps** [comment]
It's not a hard blocker. The input validation checks live in the common package, so it would make sense to add tests to the common package alongside this ticket but we can also test the fixes e2e no problem. I'll go ahead and remove the dependency in favor of this note that those two tickets would be good candidates to assign in parallel.

**Comment by alecps** [comment]
Adding some specifics here

In phone-number-privacy/common/input-validation.ts 

- [x] hasValidQueryPhoneNumberParam() should do more than just check if the field exists in the request object. We should be able to handle any type of input here but in particular we should reject requests that provide an unblinded phone number in this field as it's an easy mistake to make. (highest priority) 

- [x] hasValidUserPhoneNumberParam() if possible should actually reject input that isn't in a valid phone number format. Once we have this, we should add it to hasValidContractPhoneNumbersParam() and fix the typo (it should be 'Contact' not 'Contract') 

- [x] hasValidPhoneNumberHash() is a duplicate of phoneNumberHashIsValidIfExists(). We should only be using one of these. Currently hasValidPhoneNumberHash() is only used in the matchmaking flow so it may be easier to deprecate. We should also try to find a way to validate whether a 'phone number hash' is valid or not. 

- [x] Testing --> add unit tests to the combiner and signer index.test.ts files. We should also probably add an e2e test to confirm that the hasValidQueryPhoneNumberParam() bug is fixed and update or add any other tests that feel appropriate.
